### PR TITLE
cppcheck: 2.7 -> 2.7.3

### PR DESCRIPTION
**Pull request by bjornfor**
###### Description of changes
Fix packaging of `cppcheck-htmlreport` and bump to the latest version.

And switch from sourceforge to github, as upstream hasn't updated the
sourceforge pages yet. (I checked the download links on the sourceforge
page and they link to https://github.com/danmar/cppcheck, so that should
be fine.)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/analysis/cppcheck/default.nix [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+  src = fetchFromGitHub {
+    owner = "danmar";
+    repo = "cppcheck";
+    rev = version;
+    sha256 = "0bwk89nkq67nphplb24daxvg75pv9bgh0kcqr2samhpzmjpvzxm5";
   };
 
-  buildInputs = [ pcre ] ++ lib.optionals withZ3 [ z3 ];
+  buildInputs = [ pcre
+    (python3.withPackages (ps: [ps.pygments]))
```
Comment (risicle) [inline pkgs/development/tools/analysis/cppcheck/default.nix:15]: Not sure if this construction is totally necessary - doesn't just including `python3.pkgs.pygments` achieve the same thing?

Comment (risicle) [reply pkgs/development/tools/analysis/cppcheck/default.nix:15]: Ok no, having dug a bit, no it doesn't and I see why you did it this way.

**Comment by risicle** [comment]
Suggest adding `doCheck = true;` as the tests pass for me, nixos x86_64 & macos 10.15.

**Comment by bjornfor** [comment]
> Suggest adding `doCheck = true;` as the tests pass for me, nixos x86_64 & macos 10.15.

Good idea, done.

I also tested it on aarch64, through emulation (`boot.binfmt.emulatedSystems = [ "aarch64-linux" ]`).

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Pull request merged**